app/models/page.rb

```diff
@@ -57,6 +57,24 @@ def meta_tags
     tag_names << plugin_names
   end
 
+  def dup
+    clone = super
+    clone.primary_image = nil
+    clone.slug = nil
+    clone
+  end
+
+  def number_of_pages_with_matching_title
+    Page.where(title: title).count
+  end
+
+  def slug_candidates
+    [
+      :title,
+      [:title, :number_of_pages_with_matching_title]
+    ]
+  end
```

Comment:
Haven't thought about this a lot, but from product perspective I think if this allows campaigner to create two pages with the same public-facing title easily, that's good. I'm unsure how this will display in the campaigner view, however -- will there be any way to distinguish the pages easily (given all you currently see is the page title?) We might want to create an optional Page Notes or something and display that, too.